### PR TITLE
added Discussion Model with API

### PR DESCRIPTION
**Pull request by JaneGH**

### CODE REVIEW

File: src/controllers/discussionController.ts [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+import {
+  BadRequestError,
+  NotFoundError,
+  UnauthenticatedError,
+} from "../errors";
+import Joi from "joi";
+import mongoose from "mongoose";
+import { IUser } from "../models/User";
+
+const discussionJoiSchema = Joi.object({
```
Comment (smart-niggs) [inline src/controllers/discussionController.ts]: you may have a validation file, 
either a validation.ts for all validations or a discussionValidation.ts for just this.

File: src/controllers/discussionController.ts [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+  title: Joi.string().required(),
+  book: Joi.string().required(),
+  content: Joi.string().required(),
+  date: Joi.date().iso().required(),
+  participants: Joi.array().items(Joi.string()).optional(),
+  meetingLink: Joi.string().uri().required(),
+  createdBy: Joi.string().required(),
+});
+
+interface IGetDiscussionsQuery {
```
Comment (smart-niggs) [inline src/controllers/discussionController.ts]: you can create an interface file to store all your interfaces

File: src/controllers/discussionController.ts [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+    return next(error);
+  }
+};
+
+const getDiscussion = async (
+  req: Request<{ discussionId: string }>,
+  res: Response,
+  next: NextFunction
+) => {
+  const { discussionId } = req.params;
```
Comment (smart-niggs) [inline src/controllers/discussionController.ts:64]: you can use validation whenever any input is received from the user, in this case: discussionId

File: src/controllers/discussionController.ts [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+interface IJoinDiscussionBody {
+  user: mongoose.Schema.Types.ObjectId;
+}
+
+const getAllDiscussions = async (
+  req: Request<{}, {}, {}, IGetDiscussionsQuery & { bookId?: string }>,
+  res: Response,
+  next: NextFunction
+) => {
+  const { search, sort, bookId } = req.query;
```
Comment (smart-niggs) [inline src/controllers/discussionController.ts:27]: you can use validation whenever any input is received from the user,

File: src/controllers/discussionController.ts [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+  next: NextFunction
+) => {
+  const { discussionId } = req.params;
+  const user: IUser | undefined = req.user;
+
+  if (!user) {
+    return next(new UnauthenticatedError("User is not authenticated"));
+  }
+
+   const discussion = await Discussion.findById(discussionId);
```
Comment (smart-niggs) [inline src/controllers/discussionController.ts:119]: you can also do sth like:
`const discussion = await Discussion.findOne({
  id: ObjectId(discussionId),
  createdBy: user.userId
});

if (!discussion) {
  return next(new NotFoundError(`No discussion with id ${discussionId} or user is not authorized to delete`));
}

try {
  await discussion.deleteOne();
  res.status(StatusCodes.OK).send();
} catch (error) {
  return next(error);
}`

File: tsconfig.tsbuildinfo [hunk 6]
```diff
@@ -0,0 +1 @@
+{"root":["./src/app.ts","./src/server.ts","./src/controllers/bookcontroller.ts","./src/controllers/commentcontroller.ts","./src/controllers/discussioncontroller.ts","./src/controllers/maincontroller.ts","./src/controllers/usercontroller.ts","./src/db/connect.ts","./src/errors/bad-request.ts","./src/errors/custom-api.ts","./src/errors/index.ts","./src/errors/not-found.ts","./src/errors/unauthenticated.ts","./src/middleware/authentication.ts","./src/middleware/error-handler.ts","./src/models/book.ts","./src/models/comment.ts","./src/models/discussion.ts","./src/models/user.ts","./src/routes/bookroutes.ts","./src/routes/commentroutes.ts","./src/routes/discussionroutes.ts","./src/routes/mainrouter.ts","./src/routes/userroutes.ts","./src/services/mailjetservice.ts","./src/types/express.d.ts","./src/utils/swagger.ts"],"version":"5.6.3"}
```
Comment (smart-niggs) [inline tsconfig.tsbuildinfo]: what does this do?

**Review by smart-niggs — COMMENTED** [review]
Great work @JaneGH ,
take a look at the review comments

File: src/controllers/discussionController.ts (hunk 5 above)
Comment (JaneGH) [reply src/controllers/discussionController.ts:119]: In this case, I don't like that we're not clear on whether the user is unauthorized or if there's no discussion with the id ${discussionId}. Can I leave it like this for now?

File: src/controllers/discussionController.ts (hunk 1 above)
Comment (JaneGH) [reply src/controllers/discussionController.ts]: fixed

File: src/controllers/discussionController.ts (hunk 2 above)
Comment (JaneGH) [reply src/controllers/discussionController.ts]: done

File: src/controllers/discussionController.ts (hunk 3 above)
Comment (JaneGH) [reply src/controllers/discussionController.ts:64]: done

File: src/controllers/discussionController.ts (hunk 4 above)
Comment (JaneGH) [reply src/controllers/discussionController.ts:27]: done

File: tsconfig.tsbuildinfo (hunk 6 above)
Comment (JaneGH) [reply tsconfig.tsbuildinfo]: I deleted it. I don’t know why or when it was added. Maybe it was added automatically.

File: src/controllers/discussionController.ts (hunk 5 above)
Comment (smart-niggs) [reply src/controllers/discussionController.ts:119]: okay, you can bring this up when we sync

File: src/controllers/discussionController.ts (hunk 2 above)
Comment (smart-niggs) [reply src/controllers/discussionController.ts]: is this done? I can see the interface

**Review by smart-niggs — APPROVED** [review]
You can look at the unresolved comments and work on them in your next PR

**Pull request merged**